bigframes/core/indexes/base.py

```diff
@@ -90,9 +90,12 @@ def __new__(
         # TODO: Support more index subtypes
         from bigframes.core.indexes.multi import MultiIndex
 
-        klass = MultiIndex if len(block._index_columns) > 1 else cls
-        # TODO(b/340893286): fix type error
-        result = typing.cast(Index, object.__new__(klass))  # type: ignore
+        if len(block._index_columns) <= 1:
```

Comment:
hmmm. I'm surprised that mypy distinguishes those two syntax. Do you know why is it?

Reply:
From what I find, first, mypy can't tell the type from things like x if y else z, it will assume it's object.

Then, if we use if else, multiindex is a subclass of index, so we need to handle the index class first, otherwise it will return another type error. 

So this is the final result.